lib/extensions/src/lib/services/extension.service.ts

```diff
@@ -118,9 +118,9 @@ export class ExtensionService {
      * Gets features by key.
      *
      * @param key Key string, using dot notation
-     * @returns Features array found by key
+     * @returns Feature found by key
      */
-    getFeature(key: string): any[] {
+    getFeature<T = any[]>(key: string): T {
         const properties: string[] = Array.isArray(key) ? [key] : key.split('.');
         return properties.reduce((prev, curr) => prev && prev[curr], this.features) || [];
```

Comment:
note the ending of the line here, it returns an empty array while your `T` does not state that

Reply:
hmm the default is array of any, so in my opinion this is Ok. 

In case if the searched `features` is something different and we know that e.g. the searched feature is an object, then we can use generic for it.

will add a test for it as well

Reply:
ps. there is something wrong with the typings e.g. 

```
getFeature<T = any[]>(key: string): T {
        const properties: string[] = Array.isArray(key) ? [key] : key.split('.');
```
The `key` can only be `string`, but we are checking if it is an `array` and if it is an array, then we are adding it to another array, so we ended-up with type for `properties` set to nested array - `string[][]` 

i.e. 
<img width="1032" alt="Screenshot 2023-01-30 at 13 02 50" src="https://user-images.githubusercontent.com/13885344/215471900-1b590da3-abc2-450e-af84-a2a9d933f57c.png">

Reply:
probably was some sort of the bug